sky/jobs/state.py

```diff
@@ -728,14 +729,22 @@ def get_jobs_to_check_status(job_id: Optional[int] = None) -> List[int]:
             LEFT OUTER JOIN job_info
             ON spot.spot_job_id=job_info.spot_job_id
             WHERE (
+                -- non-legacy jobs that are not DONE
                 (job_info.schedule_state IS NOT NULL AND
                  job_info.schedule_state IS NOT ?)
                 OR
-                (job_info.schedule_state IS NULL AND
+                -- legacy or DONE jobs that are in non-terminal status
+                ((-- legacy jobs
+                  job_info.schedule_state IS NULL OR
+                  -- DONE jobs
```

Comment:
```suggestion
                  -- DONE jobs - The legacy job's schedule state may be set to DONE during a previous status update
```

Reply:
This is incorrect, it's not for handling legacy jobs. Tried to clarify in the other comment.